spring-cloud-dataflow-admin-starter/src/main/java/org/springframework/cloud/dataflow/admin/controller/FieldValueCounterController.java

```diff
@@ -0,0 +1,140 @@
+import org.springframework.hateoas.ExposesResourceFor;
+import org.springframework.hateoas.PagedResources;
+import org.springframework.hateoas.mvc.ResourceAssemblerSupport;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Allows interaction with Field Value Counters.
+ *
+ * @author Eric Bottard
+ */
+@RestController
+@RequestMapping("/metrics/field-value-counters")
+@ExposesResourceFor(FieldValueCounterResource.class)
+public class FieldValueCounterController {
+
+	private final FieldValueCounterRepository repository;
+
+	@Autowired
+	public FieldValueCounterController(FieldValueCounterRepository repository) {
+		this.repository = repository;
+	}
+
+	private DeepResourceAssembler deepAssembler = new DeepResourceAssembler();
+
+	private ShallowResourceAssembler shallowAssembler = new ShallowResourceAssembler();
+
+	/**
+	 * Retrieve information about a specific counter.
+	 */
+	@RequestMapping(value = "/{name}", method = RequestMethod.GET)
+	public FieldValueCounterResource display(@PathVariable("name") String name) {
+		FieldValueCounter counter = repository.findOne(name);
+		if (counter == null) {
+			throw new MetricsMvcEndpoint.NoSuchMetricException(name);
+		}
+		return deepAssembler.toResource(counter);
+	}
+
+	/**
+	 * Delete (reset) a specific counter.
+	 */
+	@RequestMapping(value = "/{name}", method = RequestMethod.DELETE)
+	@ResponseStatus(HttpStatus.OK)
+	protected void delete(@PathVariable("name") String name) {
+		FieldValueCounter counter = repository.findOne(name);
+		if (counter == null) {
+			throw new MetricsMvcEndpoint.NoSuchMetricException(name);
+		}
+		repository.reset(name);
+	}
+
+	/**
+	 * List Counters that match the given criteria.
+	 */
+	@RequestMapping(value = "", method = RequestMethod.GET)
+	public PagedResources<? extends MetricResource> list(
+			PagedResourcesAssembler<String> pagedAssembler) {
+		List<String> names = new ArrayList<>(repository.list());
+		return pagedAssembler.toResource(new PageImpl<>(names), shallowAssembler);
+	}
+
+	/**
+	 * Knows how to assemble {@link MetricResource} out of simple String names
+	 */
+	private static class ShallowResourceAssembler extends
+			ResourceAssemblerSupport<String, MetricResource> {
+
+		private ShallowResourceAssembler() {
+			super(FieldValueCounterController.class, MetricResource.class);
+		}
+
+		@Override
+		public MetricResource toResource(String name) {
+			return createResourceWithId(name, name);
+		}
+
+		@Override
+		protected MetricResource instantiateResource(String name) {
+			return new MetricResource(name);
+		}
+	}
+
+	/**
+	 * Knows how to assemble {@link FieldValueCounterResource}s out of counter {@link FieldValueCounter}s.

```

Comment:
I think it can be better this way:

```
Knows how to assemble {@link FieldValueCounterResource} out of  {@link FieldValueCounter}.
``

Removal of plural and extra word `counter`.